src/presentation/ui/layout/sidebar.templ

```diff
@@ -0,0 +1,125 @@
+		{Label: "Mappings", Icon: "ph-graph", Link: "/mappings/"},
+		{Label: "SSL Certificate", Icon: "ph-lock", Link: "/_/#/ssl/"},
+		{Label: "Marketplace", Icon: "ph-storefront", Link: "/_/#/marketplace/"},
+		{Label: "Apps", Icon: "ph-circles-four", Link: ""},
+		{Label: "Domains", Icon: "ph-globe", Link: ""},
+		{Label: "Backups", Icon: "ph-cloud-arrow-up", Link: ""},
+		{Label: "Web Servers", Icon: "ph-browsers", Link: ""},
+		{Label: "Metrics", Icon: "ph-chart-bar", Link: ""},
+		{Label: "Logs", Icon: "ph-log", Link: ""},
+		{Label: "Image Optimizer", Icon: "ph-image-square", Link: ""},
+		{Label: "Security", Icon: "ph-shield-checkered", Link: ""},
+		{Label: "Terminal", Icon: "ph-terminal-window", Link: ""},
+	}
+
+	for itemIndex, item := range sidebarItems {
+		if item.Link != currentUrl {
+			continue
+		}
+
+		sidebarItems[itemIndex].AdditionalClasses = " text-speedia-500"
+		break
+	}
+
+	return sidebarItems
+}
+
+templ Sidebar(currentUrl string) {
+	<!-- Sidebar JavaScript -->
+	<script defer type="text/javascript">
+	function clearSession() {
+		document.cookie = "os-access-token=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;";
+		window.location.href = "/_/#/login";
+	}
+	function openSettings() {
+		window.location.href = "/_/#/settings";
+	}
+    function scrollToTop() {
+		const menu = document.getElementById("sidebarMenu");
+		menu.scrollTo({ top: 0, behavior: 'smooth' });
+	}
+    function toggleScrollButton() {
+		const menu = document.getElementById("sidebarMenu");
+		const scrollButton = document.getElementById("scrollButton");
+		if (menu.scrollTop > 0) {
+			scrollButton.style.display = 'flex';
+		} else {
+			scrollButton.style.display = 'none';
+		}
+	}
+	</script>
+	<!-- Sidebar HTML -->
+	<sidebar class="bg-os-900 flex h-screen w-30 flex-col items-center">
+		<div class="my-7">
+			<a
+				href="/_/#/overview"
+				class="mx-auto mb-2 flex w-[72px] max-w-[72px] items-center justify-center"
+			>
+				<img
+					src="/assets/os-logo.svg"
+					alt="Speedia OS"
+				/>
+			</a>
+			<p class="text-center text-xs">v{ infraEnvs.SpeediaOsVersion }</p>
+		</div>
+		<button
+			type="button"
+			onclick="clearSession()"
+			class="hover:bg-speedia-500 flex items-center justify-center rounded-md border border-neutral-500 bg-transparent px-3 py-1.5 text-center text-xs text-neutral-400 transition-all hover:border-transparent hover:text-white"
+		>
+			<i class="ph-bold ph-sign-out pr-1 pt-1"></i>
+			logout
+		</button>
+		<button
+			id="scrollButton"
+			onclick="scrollToTop()"
+			class="hidden rounded-full border border-neutral-500 bg-transparent transition-all hover:border-transparent hover:bg-speedia-500 p-2 mt-2"
+		>
+			<i class="ph-duotone ph-arrow-fat-up block"></i>
+		</button>
+		<div id="sidebarMenu" onscroll="toggleScrollButton()" class="mt-2 flex-1 w-full overflow-y-auto scrollbar-hide hover:scrollbar-show">
+			<ul>
+				for _, item := range sidebarItemsGetter(currentUrl) {
+					if item.Link != "" {
```

Comment:
If there is no link, skip it instead of putting everything inside an if block. That's dev 101 dude.

Reply:
Use the for loop of sidebarItemsGetter to remove empty Links itens.

Reply:
There isn't a very user-friendly way to do this other than finding the index and going through that entire lengthy process. To avoid this, I decided to create a slice containing the available ones and populate it using a for loop.